### PR TITLE
set state tcp connection

### PR DESCRIPTION
**Pull request by johnk3r**
Rule to detect the use of the "SetTcpEntry" API, commonly used by TA to disable EDR.

### CODE REVIEW

File: host-interaction/network/connectivity/set-state-tcp-connection.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+rule:
+  meta:
+    name: set state tcp connection
```
Comment (williballenthin) [inline host-interaction/network/connectivity/set-state-tcp-connection.yml:3]:
```suggestion
    name: set TCP connection state
```

**Review by williballenthin — CHANGES_REQUESTED** [review]
logic looks great, recommend tweaking the rule name

**Comment by mr-tz** [comment]
please rename to `set-tcp-connection-state.yml` then good to merge

File: host-interaction/network/connectivity/set-tcp-connection-state.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+rule:
+  meta:
+    name: set TCP connection state
+    namespace: host-interaction/network/connectivity
+    authors:
+      - "@johnk3r"
+    description: The SetTcpEntry function sets the state of a TCP connection.
+    scope: function
+    att&ck:
+      - Defense Evasion::Impair Defenses [T1562]
```
Comment (mr-tz) [inline host-interaction/network/connectivity/set-tcp-connection-state.yml:10]: Can we be sure this is always the technique here?

Comment (johnk3r) [reply host-interaction/network/connectivity/set-tcp-connection-state.yml:10]: MSFT describes the following:

1 - **The SetTcpEntry function sets the state of a TCP connection.**
2 - **Currently, the only state to which a TCP connection can be set is MIB_TCP_STATE_DELETE_TCB.**

So in my understanding yes. Do you have any other experience with this?

Comment (mr-tz) [reply host-interaction/network/connectivity/set-tcp-connection-state.yml:10]: I see, should we then add `number: 12 = MIB_TCP_STATE_DELETE_TCB` to the logic?

Comment (johnk3r) [reply host-interaction/network/connectivity/set-tcp-connection-state.yml:10]: Nice ideia!


image

What do you think?

Comment (mr-tz) [reply host-interaction/network/connectivity/set-tcp-connection-state.yml:10]: looks good, please add it to the rule and then we're g2g

**Review by mr-tz — APPROVED** [review]
thank you!

**Pull request merged**